plugins/volumeFilter.go

```diff
@@ -175,74 +175,95 @@ func (f *volumeFilter) Apply(ops []txnbuild.Operation, sellingOffers []hProtocol
 }
 
 func volumeFilterFn(dailyOTB *VolumeFilterConfig, dailyTBBAccumulator *VolumeFilterConfig, op *txnbuild.ManageSellOffer, baseAsset hProtocol.Asset, quoteAsset hProtocol.Asset, lp limitParameters) (*txnbuild.ManageSellOffer, error) {
-	isSell, e := utils.IsSelling(baseAsset, quoteAsset, op.Selling, op.Buying)
+	isFilterApplicable, e := offerSameTypeAsFilter(dailyOTB, op, baseAsset, quoteAsset)
 	if e != nil {
-		return nil, fmt.Errorf("error when running the isSelling check for offer '%+v': %s", *op, e)
+		return nil, fmt.Errorf("could not compare offer and filter: %s", e)
 	}
 
-	sellPrice, e := strconv.ParseFloat(op.Price, 64)
+	if !isFilterApplicable {
+		// ignore filter so return op directly
+		return op, nil
+	}
+
+	offerPrice, e := strconv.ParseFloat(op.Price, 64)
```

Comment:
not sure if op.Price is inverted in the case of a buy because of the complexities of how manage sell offer works. It's complicated and not something we need to get right in this PR.

but this is something we should look at when working on the tests for the buy side. Can you add a TODO DS comment here so we remember to get this right in the next PR?